### PR TITLE
Theme Support

### PR DESCRIPTION
**Pull request by xsc**
This PR implements theme support.

A theme with name `awesome` resides in `codox/themes/awesome` on the classpath. Within, it provides at least the file `theme.edn` with enlive transformations<sup>1</sup>:

```clojure
{:transforms
 [[:head]
  [:append
   [:script {:src ...}]]]}
```

Additionally, it can contain the directory `public` whose contents will be copied directly to the codox output directory. The theme would be activated within the `project.clj` by setting:

```clojure
:codox {:theme :awesome}
```

See #135.

---

__Notes__

The last two commits are to-be-discussed:

- 86b51c6 implements support for combination of themes since there might be non-styling themes (like KLIPSE support outlined in #137) that could be added on top of a "real" theme.
- 67ecd6e allows themes to set default metadata, e.g. enable Markdown by default. I feel like the verbosity of the code and the fact that it is run at the end of the processing pipeline instead of the beginning (where it arguably belongs) should be ground enough to remove it. Still, as I said, to-be-discussed.

[cpath-clj](https://github.com/xsc/cpath-clj) is used to collect the theme files in `public`. If you want to avoid another dependency we could include the single namespace it consists of, plus the `filenames-in-jar` function from `clojure.java.classpath`, directly. Heeding the respective licenses, of course.

---

I prepared an example theme here: https://github.com/xsc/codox-theme-rdash

### CODE REVIEW

**Comment by weavejester** [comment]
If I had known about cpath-clj a year ago, I wouldn't have written [Resauce](https://github.com/weavejester/resauce)! But since I have written it, can we replace `cpath-clj.core/resources` with `resauce.core/resources`? It'll make it easier for me to deal with any potential issues.

Rather than `:transformations`, can we use a `:transforms` key instead to match up with the Codox configuration?

I like the idea of 36988bb. Seems sensible, and is one of the advantage of having a data-based design.

Regarding ed53d04, I think let's save that for a future PR. I'm not convinced config options should be part of themes, which seem to me like they should be purely aesthetic.

**Comment by weavejester** [comment]
Also, `codox.writer.themes` should probably be `codox.writer.html.themes`, since the themes only apply to the HTML writer.

**Comment by xsc** [comment]
I added some commits addressing your comments. Once everything is fine, I'd like to squash them into about three total.

File: codox/project.clj [hunk 1]
```diff
@@ -9,5 +9,7 @@
                  [org.clojure/clojurescript "1.7.189"]
                  [hiccup "1.0.5"]
                  [enlive "1.1.6"]
+                 [resauce "0.1.0"]
+                 [cpath-clj "0.1.2"]
```
Comment (weavejester) [inline codox/project.clj]: This dep can be removed now

**Comment by weavejester** [comment]
It looks really good. I'm going to release Codox 0.9.7 with the syntax highlighting changes (because they're a nice final addition to the 0.9.x series), then tomorrow evening I'll have a play around with this and likely release Codox 0.10.0.

**Comment by weavejester** [comment]
Could you possibly add a simple "test" theme to the example project? Just so we have something to visually test the theme system.

**Comment by xsc** [comment]
Added test theme (in the `:theme` profile), squashed commits.

I also noticed that, by using `clojure.edn` for EDN parsing, codox is no longer compatible with Clojure 1.3.0. Is that a problem?

**Comment by weavejester** [comment]
I think it's fine to drop Clojure 1.3.0 support for Codox 0.10.0, but it should still work with Clojure 1.5.1. Can you set the example project to use Clojure 1.5.1 rather than 1.8.0, so we can test it works with the new minimum version?

**Comment by xsc** [comment]
I adjusted the Clojure version in the example project.

**Comment by weavejester** [comment]
Perfect. Can you squash the last two commits? 	i.e. ae95fc4 and fe6de90.

**Comment by xsc** [comment]
Done. :)

**Pull request merged**